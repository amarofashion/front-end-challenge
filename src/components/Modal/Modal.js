import React from 'react';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Creators as OverlayActionsCreator } from '../../store/ducks/overlay';
import { Creators as QuickViewActionsCreator } from '../../store/ducks/quickview';

import '../../styles/components/Modal.scss';

import Button from '../Button';

const Modal = ({
  children,
  className,
  overlayActions,
  quickViewActions,
}) => {
  const { toggleOverlay } = overlayActions;
  const {
    toggleQuickView,
    sizeProductQuickView,
    errorSizeBullets,
  } = quickViewActions;

  const handleClose = () => {
    toggleOverlay(false);
    toggleQuickView(false);
    errorSizeBullets(false);
    sizeProductQuickView(false);
  };
  return (
    <div className={`am-modal ${className}`}>
      <Button
        className="am-modal__close"
        onClick={() => handleClose()}
      />
      {children}
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  overlayActions: bindActionCreators(OverlayActionsCreator, dispatch),
  quickViewActions: bindActionCreators(
    QuickViewActionsCreator,
    dispatch,
  ),
});

export default connect(null, mapDispatchToProps)(Modal);
