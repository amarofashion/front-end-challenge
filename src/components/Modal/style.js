import styled from 'styled-components';
import breakpoint from 'styled-components-breakpoint';

import colors from '../../theme/colors';
import Icon from '../Icon';

export const StyledBackground = styled.div`
  z-index: 100;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(31, 45, 48, 0.88); /* outerspace rgba */
`;

export const StyledWrapper = styled.div`
  background: ${colors.alabaster};
  position: relative;
  width: 100%;
  height: calc(100% - 30px);
  top: 50px;
  left: 0;

  ${breakpoint('tablet')`
    position: fixed;
    width: auto;
    height: auto;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  `}
`;

export const StyledButton = styled.button`
  position: absolute;
  top: -40px;
  right: 0;
  height: 30px;
  width: 30px;
  text-align: center;
  border: none;
  background: none;
  padding: 0;
  margin: 0;
  color: ${colors.light};

  &:hover {
    cursor: pointer;
  }
`;

export const StyledIcon = styled(Icon)`
  color: ${colors.alabaster};
`;

export const StyledScroll = styled.div`
  overflow-y: auto !important;
  height: 100%;

  ${breakpoint('tablet')`
    max-height: 600px;
  `};
`;
