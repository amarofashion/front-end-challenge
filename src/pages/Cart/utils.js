import { translateToNumber } from '../../utils/number';

export function getPrices(list) {
  const result = {
    subTotal: 0,
    totalDiscount: 0,
    total: 0,
  };

  list.forEach(({ onSale, actualPrice, regularPrice, quantity }) => {
    const actualPriceValue = translateToNumber(actualPrice) * quantity;
    const regularPriceValue = translateToNumber(regularPrice) * quantity;

    if (onSale) {
      result.totalDiscount += regularPriceValue - actualPriceValue;
      result.total += actualPriceValue;
    } else {
      result.total += regularPriceValue;
    }

    result.subTotal += regularPriceValue;
  });

  return result;
}
