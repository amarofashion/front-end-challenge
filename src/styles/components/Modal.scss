@import '../settings/index.scss';


.am-modal{
  @include z-index(modal);
  @extend %center;
  position: fixed;
  background-color: $color-white;
  top: 0;
  left: 0;
  width: 95vw;
  height: 90vh;
  left: 50%;
  top: 50%;
  transform: translate(-200%, -50%);
  visibility: hidden;
  opacity: 0;
  transition: transform 300ms linear;
  &.is--active{
    opacity: 1;
    visibility: visible;
    transform: translate(-50%, -50%);
  }
  &__close{
    @include rem(width, $gap-greatest);
    @include rem(height, $gap-greatest);
    @include icon-more(20px, 1px, $color-black);
    margin-left: auto;
    position: relative;
    display: block;
    outline: none;
    transform: rotate(45deg);
  }
}