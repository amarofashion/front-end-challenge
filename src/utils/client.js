import axios from 'axios';

const axiosInstance = axios.create({
  baseURL: 'https://testapi.io/api/sergiokopplin',
});

export { axiosInstance };
