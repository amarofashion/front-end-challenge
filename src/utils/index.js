const replaceValue = value => {
  return value
    .trim()
    .toLowerCase()
    .replace(/[^\w]+/g, '-');
};

const friendlyURL = (name = '', color = '', style = '') => {
  return replaceValue(name)
    .concat(`-${replaceValue(color)}`)
    .concat(`-${style}`);
};

export default friendlyURL;
