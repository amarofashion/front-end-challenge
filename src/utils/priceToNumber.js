const priceToNumber = price =>
  Number(price.replace(/[^0-9.-]+/g, ''));

export default priceToNumber;
