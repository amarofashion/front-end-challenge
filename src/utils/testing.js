import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { initialState } from '../store/reducers';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore(initialState);

// eslint-disable-next-line react/prop-types
export const WithStore = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

// eslint-disable-next-line react/prop-types
export const WithRouter = ({ children }) => <Router>{children}</Router>;
