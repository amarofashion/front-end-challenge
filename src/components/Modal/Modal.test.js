import React from 'react';
import { mount } from 'enzyme';

import Modal from '.';
import { WithStore } from '../../utils/testing';

describe('<Modal />', () => {
  describe('with default props', () => {
    it('shoud match snapshot with default props', () => {
      const wrapper = mount(
        <WithStore>
          <Modal>
            <div>test</div>
          </Modal>
        </WithStore>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
