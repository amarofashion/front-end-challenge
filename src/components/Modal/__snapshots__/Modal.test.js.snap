// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> with default props shoud match snapshot with default props 1`] = `
<WithStore>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Modal />
  </Provider>
</WithStore>
`;
